src/main/java/io/github/artsok/extension/ParameterizedRepeatedExtension.java

```diff
@@ -209,8 +227,10 @@ private ArgumentsProvider instantiateArgumentsProvider(Class<? extends Arguments
 
         @Override
         public boolean hasNext() {
-            if (historyExceptionAppear.stream().anyMatch(ex -> ex) && currentIndex < totalRepeats) {
-                return historyExceptionAppear.stream().anyMatch(ex -> ex) && currentIndex < totalRepeats;
+            if (!historyExceptionAppear.isEmpty()
+                    && historyExceptionAppear.get(historyExceptionAppear.size() - 1)
+                    && currentIndex < totalRepeats) {
+                return true;
```

Comment:
This fixes an IndexOutOfBoundsException when the last iteration of a parameterized test fails at least once but then succeeds within the repetitions